src/main/java/com/dumbdogdiner/stickyapi/common/user/StickyUser.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020-2021 DumbDogDiner <dumbdogdiner.com>. All rights reserved.
+ * Licensed under the MIT license, see LICENSE for more information...
+ */
+package com.dumbdogdiner.stickyapi.common.user;
+
+import com.dumbdogdiner.stickyapi.common.cache.Cacheable;
+import lombok.Getter;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+/**
+ * A base user class to eliminate some code duplication
+ */
+public abstract class StickyUser implements Cacheable {
```

Comment:
Did we ever agree on how we would implement this? Thought for quite a long time it makes much more sense to just pass plugin-specific data about.

Reply:
Do correct me if I'm wrong.